CMakeLists.txt

```diff
@@ -171,10 +192,22 @@ if(USE_CUDA)
   set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
   message(STATUS "Configured CUDA host compiler: ${CMAKE_CUDA_HOST_COMPILER}")
 
+  if(POLICY CMP0104)
+    cmake_policy(SET CMP0104 NEW)
+  endif()
+
+  if(GPU_COMPUTE_VER OR NOT CMAKE_CUDA_ARCHITECTURES)
```

Comment:
Since we always have cmake 3.18+ which has `CMAKE_CUDA_ARCHITECTURES` why don't we transform the value in `GPU_COMPUTE_VER` to `CMAKE_CUDA_ARCHITECTURES`?

Roughly the check would be:
```cmake
if(DEFINED GPU_COMPUTE_VER  AND NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
  transform....
endif()

Reply:
I was going for minimal change but that's a better end result anyways, especially after stripping out the `< 3.18` `else` branches